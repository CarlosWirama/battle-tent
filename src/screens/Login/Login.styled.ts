import styled from 'styled-components';
import { Centered } from '../../components/basics';
import BackgroundBlur from '../../assets/images/title-background-blur.jpg';

export const LayoutContainer = styled(Centered)`
  background-image: url(${BackgroundBlur});
  height: 100vh;
`;

export const LogoHeader = styled.header`
  flex: 1;
  display: flex;
  align-items: flex-end;
`;
