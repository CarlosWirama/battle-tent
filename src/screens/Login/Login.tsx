import React, { useContext } from 'react';
import Auth from '../../utils/auth';
import { NavigationContext, ScreenState } from '../../navigation';
import { Input, Button } from '../../components/basics';
import { LayoutContainer } from './Login.styled';

export default function Login() {
  const navigate = useContext(NavigationContext);

  function onLogin() {
    Auth.authenticate(() => {
      navigate(ScreenState.Lobby);
    });
  }

  return (
    <LayoutContainer>
      <header>
        <div>Welcome to</div>
        <h1>Battle Tent</h1>
      </header>
      <main>
        <h5>What is your display name?</h5>
        <Input autoFocus placeholder="...Trainer123" />
        <Button onClick={onLogin}>Enter</Button>
      </main>
    </LayoutContainer>
  );
}
