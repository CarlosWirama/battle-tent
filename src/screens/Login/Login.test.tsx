import React from 'react';
import { render } from '@testing-library/react';
import Login from './Login';
import { BrowserRouter } from 'react-router-dom';

describe('Login screen', () => {
  it('should render', () => {
    const { asFragment } = render(
      <BrowserRouter>
        <Login />
      </BrowserRouter>
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
