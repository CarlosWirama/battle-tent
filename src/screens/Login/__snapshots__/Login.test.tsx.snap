// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login screen should render 1`] = `
<DocumentFragment>
  <div>
    <header>
      <h1>
        Welcome to Battle Tent
      </h1>
    </header>
    <div>
      <h5>
        What is your display name?
      </h5>
      <input
        value=""
      />
      <button
        class="MuiButtonBase-root MuiButton-root MuiButton-text"
        tabindex="0"
        type="button"
      >
        <span
          class="MuiButton-label"
        >
          Enter
        </span>
        <span
          class="MuiTouchRipple-root"
        />
      </button>
    </div>
  </div>
</DocumentFragment>
`;
